packages/core/admin/ee/server/services/review-workflows/workflows/content-types.js

```diff
@@ -28,12 +28,17 @@ module.exports = ({ strapi }) => {
      * @param {Array<string>} options.srcContentTypes - The content types assigned to the previous workflow
      * @param {Array<string>} options.destContentTypes - The content types assigned to the new workflow
      * @param {Workflow.Stage} options.stageId - The new stage to assign the entities to
+     * @param {boolean} force - Override validations to force the migration (e.g. on workflow deletion)
      */
-    async migrate({ srcContentTypes = [], destContentTypes, stageId }) {
+    async migrate({ srcContentTypes = [], destContentTypes, stageId }, force = false) {
       // Workflows service is using this content-types service, to avoid an infinite loop, we need to get the service in the method
       const workflowsService = getService('workflows', { strapi });
+      const workflowsValidationService = getService('review-workflows-validation', { strapi });
       const { created, deleted } = diffContentTypes(srcContentTypes, destContentTypes);
 
+      if (!force) {
+        await workflowsValidationService.validateWorkflowCount();
+      }
```

Comment:
Is it necessary to validate the workflow count at this point? Imo the workflow service itself must know about its validations, but its not necessary for the content-type migrations itself to do so.

By looking at the code it seems all workflow service methods using this migrate are already validating the workflow count.

Reply:
This is true, but any part of the app can load this service and use the migrate function outside of the workflow service